### PR TITLE
Reduce number of sql queries from 10k to 500 in pinot-integration-tests module

### PR DESCRIPTION
**Pull request by jackjlli**
This PR:
* removes queries with empty response.
* reduces the number of sql queries from 10k to 500 in pinot-integration-tests module.

### CODE REVIEW

**Review by Jackie-Jiang — COMMENTED** [review]
Can we make the sql queries one-to-one mapping with the pql queries?

**Comment by jackjlli** [comment]
> Can we make the sql queries one-to-one mapping with the pql queries?

It's a good idea but the thing is that there are some restrictions in SQL comparisons, e.g. in SQL we skip the comparison of MV columns. So if we make it one to one mapping, the number of queries will be reduced. 
Plus, SQL requires the order by column to be appeared in the selection clause, which requires us to modify the original queries. And the selection queries without order by are missing in the original query files as well, as the logic of comparing selection w/o order by is missing in the code.

**Comment by siddharthteotia** [comment]
> > Can we make the sql queries one-to-one mapping with the pql queries?
> 
> It's a good idea but the thing is that there are some restrictions in SQL comparisons, e.g. in SQL we skip the comparison of MV columns. So if we make it one to one mapping, the number of queries will be reduced.
> Plus, SQL requires the order by column to be appeared in the selection clause, which requires us to modify the original queries. And the selection queries without order by are missing in the original query files as well, as the logic of comparing selection w/o order by is missing in the code.

@jackjlli , SQL doesn't require the order by column to be present in the selection clause. May be you meant GROUP BY since that requires the query change. Also, I don't think we should skip comparing MV columns for SQL. Why can't we leverage the same comparator (that works for PQL) for SQL queries as well? Doing 1-1 test between PQL and SQL would be perfect apples to apples comparison

**Comment by mcvsubbu** [comment]
@jackjlli  for this iteration, if there is comparison missing you can choose to leave it as is with a TODO and an issue (unless you are able to add a comparator trivially).

Mark it as help wanted

**Comment by jackjlli** [comment]
@siddharthteotia I agree on that. But the restriction is not on SQL but on the code of validating the results of SQL in pinot-integration-tests module. E.g. the comparison code requires the order by columns to be appeared in the selection clause. So we may need to make the comparison code better to handle that. We can file another issue to track that, since it's not in the scope of reducing the queries.

**Comment by jackjlli** [comment]
I've filed an issue to track the issue of comparison code and label it as `help-wanted`: https://github.com/apache/incubator-pinot/issues/6193

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ClusterIntegrationTestUtils.java [hunk 1]
```diff
@@ -859,7 +859,9 @@ static void testSqlQuery(String pinotQuery, String brokerUrl, org.apache.pinot.c
     BrokerRequest brokerRequest =
         PinotQueryParserFactory.get(CommonConstants.Broker.Request.SQL).compileToBrokerRequest(pinotQuery);
     if (isSelectionQuery(brokerRequest)) { // selection
-      // TODO: compare results for selection queries, w/o order by
+      // TODO: 1. compare results for selection queries, w/o order by.
+      //       2. validate values of multi-value columns.
+      //       3. remove the restriction that order by column has to be in selection clause.
```
Comment (mcvsubbu) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ClusterIntegrationTestUtils.java:864]: Can you add the issue number here? thanks

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTest.java [hunk 2]
```diff
@@ -478,7 +479,8 @@ protected void stopKafka() {
    */
   protected long getCurrentCountStarResult()
       throws Exception {
-    return getPinotConnection().execute("SELECT COUNT(*) FROM " + getTableName()).getResultSet(0).getLong(0);
+    return getPinotConnection().execute(new Request("pql", "SELECT COUNT(*) FROM " + getTableName())).getResultSet(0)
```
Comment (mcvsubbu) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTest.java:482]: Why not make this "sql"? We will phase out pql anyway, so we can start doing this via sql all ths time?

Comment (mcvsubbu) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTest.java:482]: Ah ok, looks like this is deprecated.

**Review by mcvsubbu — APPROVED** [review]

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ClusterIntegrationTestUtils.java (hunk 1 above)
Comment (jackjlli) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ClusterIntegrationTestUtils.java:864]: I can add the issue number in the commit.

**Pull request merged**